File: LC 2641.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* replaceValueInTree(TreeNode* root) {
        root->val = 0;
        vector<TreeNode*> q{root};
        while (!q.empty()) {
            vector<TreeNode*> next;
            int sum = 0;
            for (auto& node : q) {
                if (node->left != nullptr)  {
                    next.push_back(node->left);
                    sum += node->left->val;
                }
                if (node->right != nullptr) {
                    next.push_back(node->right); 
                    sum += node->right->val;
                }
            }
            for (auto& node : q) {
                int sum_child = (node->left == nullptr ? 0 : node->left->val) + (node->right == nullptr ? 0 : node->right->val);
                if (node->left != nullptr) node->left->val = sum - sum_child;
                if (node->right != nullptr) node->right->val = sum - sum_child;
            }
            q = move(next);
        }
        return root;
    }
};

File: 3000/LC 2641.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func replaceValueInTree(root *TreeNode) *TreeNode {
    q := []*TreeNode{root}

    for len(q) > 0 {
        tmp := q
        q = nil
        sum := 0
        
        for _, node := range tmp {
            if node.Left != nil {
                q = append(q, node.Left)
                sum += node.Left.Val
            }
            if node.Right != nil {
                q = append(q, node.Right)
                sum += node.Right.Val
            }
        }

        for _, node := range tmp {
            son := 0
            if node.Left != nil {
                son += node.Left.Val
            }
            if node.Right != nil {
                son += node.Right.Val
                node.Right.Val = sum - son
            }
            if node.Left != nil {
                node.Left.Val = sum - son
            }
        }
    }

    root.Val = 0
    return root
}